0004/README.md

```diff
@@ -0,0 +1,524 @@
+# Policy Architecture for Npm Package Provenance
+
+*   Proposer: Wietse Venema ([wietse@gmail.com](mailto:wietse@gmail.com))
+*   GitHub Issue: TODO
+*   Status: [DRAFT](../README.md#meaning-of-status-codes)
+*   Original link: [Google docs](https://docs.google.com/document/d/1ulft22Nv6OXFeVRIrVfH2bK4QcRyvv0RwDt2dhJltK4/edit?usp=sharing) (see there for reviewer comments and rejected ideas)
+
+## Summary
+
+This document proposes an architecture for policy evaluation over SLSA
+provenance, for software packages that are managed with npm, the package manager
+for Node.js. When npm installs a software package and its transitive
+dependencies, it records the installed packages in a lock file
+(package-lock.json). The information that is written to these files is a primary
+input for a proposed SLSA provenance policy evaluator.
+
+With few differences, the approach presented here is expected to generalize to
+other lock file-based software environments such as Go (go.mod and go.sum), PHP
+(composer.lock), Python (Pipfile.lock), and Ruby (Gemfile.lock).
+
+## Terminology
+
+*   A *provenance policy* associates a package name with expectations for the provenance attestation properties for a collection of package versions. 
+
+*   A package version *satisfies* a provenance policy if its provenance attestation properties meet the policy expectations for the package name.
+
+## Objectives
+
+Evaluate policies over SLSA provenance for npm software packages.
+
+Non-objectives:
+
+-  Policies for what direct or indirect package dependencies (or dependency
+    versions) to expect for an npm software package; policies for vulnerability
+    scans.
+
+## Overview
+
+Here, we highlight the main architecture elements, their roles, and limitations.
+We present more detail in subsequent sections, with additional detail in the
+Appendix.
+
+### Extended software supply chain model
+
+![Extended model](extended-model.svg)
+
+The figure shows a generic supply chain for software packages, from source repository
+to installation, extended with two SLSA provenance policy evaluation points:
+
+-  When a software package is published to a package registry.
+-  When a software package is installed from a package registry.
+
+A SLSA provenance policy contains expectations for provenance, as discussed in later
```

Comment:
Sorry, it's a bit hard to understand what "expectations" are w/o first defining them upfront. The next paragraph doesn't make sense to me until I reread it again later.

Reply:
Yes, thanks!